File: packages/silo-client/src/search/search-query.ts
import type { FilterExpression } from "../query/filter-expression.js";
import type { SortTerm } from "../query/sort-term.js";

/** Everything a search accepts, all optional. `query` maps to the wire's
 *  `q`; omitting `sort` ranks by relevance instead. */
export interface SearchQuery {
  query?: string;
  where?: FilterExpression;
  sort?: SortTerm | string;
  limit?: number;
  offset?: number;
}

File: packages/silo-client-dart/lib/src/search/search_query.dart
import '../query/filter_expression.dart';

/// Everything a search accepts. [text] is the wire's `q`; without [sort],
/// results rank by relevance.
final class SearchQuery {
  const SearchQuery({this.text, this.where, this.sort, this.limit, this.offset});

  final String? text;
  final FilterExpression? where;
  final String? sort;
  final int? limit;
  final int? offset;

  SearchQuery copyWith({int? limit, int? offset}) =>
      SearchQuery(text: text, where: where, sort: sort, limit: limit ?? this.limit, offset: offset ?? this.offset);
}
